utils/src/main/java/com/cloud/utils/storage/QCOW2Utils.java

```diff
@@ -114,7 +114,8 @@ private static long getVirtualSizeFromInputStream(InputStream inputStream) throw
     public static long getVirtualSize(String urlStr) {
         try {
             URL url = new URL(urlStr);
-            return getVirtualSizeFromInputStream(url.openStream());
+            boolean isCompressed = !urlStr.endsWith("qcow2");
```

Comment:
Should we instaed check perhaps for compression extension such as tar.gz, bz2 etc? (what is somebody uploads an .img file which is a qcow2?)